File: RepoTransBench/target_projects/Python/Go/rveachkc_pymsteams/public_tests/public_cardsection_and_potentialaction_test.go
package public_tests

import (
	"reflect"
	"testing"

	"rveachkc_pymsteams/pymsteams"
)

func TestCardSectionBasicPublic(t *testing.T) {
	section := pymsteams.NewCardSection()
	if section.title("Public Section Title") != section {
		t.Error("title func should return self")
	}
	if section.payload["title"] != "Public Section Title" {
		t.Errorf("payload.title mismatch: %v", section.payload["title"])
	}
	if section.activityTitle("public activity").payload["activityTitle"] != "public activity" {
		t.Error("activityTitle payload mismatch")
	}
	if section.activitySubtitle("public subtitle").payload["activitySubtitle"] != "public subtitle" {
		t.Error("activitySubtitle payload mismatch")
	}
	if section.activityImage("https://example.com/img.png").payload["activityImage"] != "https://example.com/img.png" {
		t.Error("activityImage mismatch")
	}
	if section.activityText("public text here").payload["activityText"] != "public text here" {
		t.Error("activityText mismatch")
	}
	if section.text("greetings").payload["text"] != "greetings" {
		t.Error("text mismatch")
	}
	section.linkButton("Visit", "https://visit.com")
	actions, ok := section.payload["potentialAction"].([]interface{})
	if !ok || len(actions) == 0 {
		t.Fatal("missing potentialAction or wrong type")
	}
	first, ok := actions[0].(map[string]interface{})
	if !ok || first["name"] != "Visit" {
		t.Error("linkButton not set properly")
	}
	if section.disableMarkdown().payload["markdown"] != false {
		t.Error("disableMarkdown not working")
	}
	if section.enableMarkdown().payload["markdown"] != true {
		t.Error("enableMarkdown not working")
	}
	dumped := section.dumpSection()
	if reflect.TypeOf(dumped).Kind() != reflect.Map {
		t.Error("dumpSection did not return map")
	}
}

func TestCardSectionAddFactAndAddImagePublic(t *testing.T) {
	section := pymsteams.NewCardSection()
	section.addFact("fact_one", "value_one")
	wantFacts := []interface{}{map[string]interface{}{"name": "fact_one", "value": "value_one"}}
	if !reflect.DeepEqual(section.payload["facts"], wantFacts) {
		t.Error("addFact mismatch")
	}
	section.addFact("fact_two", "value_two")
	facts, _ := section.payload["facts"].([]interface{})
	if len(facts) != 2 {
		t.Error("addFact length error")
	}
	section.addImage("https://img-server.com/photo1.jpg", "photo title")
	images, _ := section.payload["images"].([]interface{})
	img, _ := images[0].(map[string]interface{})
	if img["title"] != "photo title" {
		t.Error("addImage title missing")
	}
	section.addImage("https://img-server.com/photo2.jpg")
	images, _ = section.payload["images"].([]interface{})
	_, ok := images[1].(map[string]interface{})
	if !ok {
		t.Fatal("second image not map")
	}
	if _, ok := images[1].(map[string]interface{})["title"]; ok {
		t.Error("title should not exist on second image")
	}
}

func TestCardSectionFactAndImageKeysPublic(t *testing.T) {
	section := pymsteams.NewCardSection()
	section.payload["facts"] = []interface{}{map[string]interface{}{"name": "init_name", "value": "init_val"}}
	section.addFact("another_name", "another_val")
	facts, _ := section.payload["facts"].([]interface{})
	if len(facts) != 2 {
		t.Error("facts len mismatch after addFact")
	}
	section.payload["images"] = []interface{}{map[string]interface{}{"image": "img_obj"}}
	section.addImage("more-img-url")
	images, _ := section.payload["images"].([]interface{})
	if len(images) != 2 {
		t.Error("images len mismatch after addImage")
	}
}

func TestPotentialActionInputsAndActionsPublic(t *testing.T) {
	pa := pymsteams.NewPotentialAction("PublicAction")
	pa.addInput("TextInput", "pub_input", "Public Input Title", false)
	inputs, _ := pa.payload["inputs"].([]interface{})
	firstmap := inputs[0].(map[string]interface{})
	if b, ok := firstmap["isMultiline"]; !ok || b != false {
		t.Error("TextInput isMultiline not set false")
	}
	pa.addInput("ChoiceInput", "pub_input2", "Public Choice Input", true)
	pa.addChoice("pub_display", "pub_value")
	inputs, _ = pa.payload["inputs"].([]interface{})
	secondmap := inputs[1].(map[string]interface{})
	if _, ok := secondmap["choices"]; !ok {
		t.Error("ChoiceInput did not set choices")
	}
	pa.addAction("CustomActionType", "ActionPublic", []string{"https://example.org"})
	acts, _ := pa.payload["actions"].([]interface{})
	actmap := acts[0].(map[string]interface{})
	if actmap["@type"] != "CustomActionType" {
		t.Error("addAction type mismatch")
	}
	pa.addAction("secondtype", "publicaction", []string{"weburl"}, "some body here")
	acts, _ = pa.payload["actions"].([]interface{})
	actmap2 := acts[1].(map[string]interface{})
	if actmap2["body"] != "some body here" {
		t.Error("addAction body param not set")
	}
}

func TestPotentialActionAddOpenURIAndExceptionsPublic(t *testing.T) {
	pa := pymsteams.NewPotentialAction("testopen")
	targets := []map[string]interface{}{{"os": "mobile", "uri": "https://other-url.com/"}}
	res, err := pa.addOpenURI("OpenOther", targets)
	if err != nil {
		t.Errorf("addOpenURI error: %v", err)
	}
	val, ok := res.payload["targets"].([]interface{})
	if !ok || len(val) == 0 {
		t.Error("addOpenURI targets missing")
	}
	_, err = pa.addOpenURI("Failer", 12345)
	if err == nil {
		t.Error("addOpenURI should error if not list")
	}
}

func TestPotentialActionDumpPublic(t *testing.T) {
	pa := pymsteams.NewPotentialAction("dumpPA")
	dumped := pa.dumpPotentialAction()
	if reflect.TypeOf(dumped).Kind() != reflect.Map {
		t.Error("dumpPotentialAction not map")
	}
}

func TestTeamsWebhookExceptionReprPublic(t *testing.T) {
	ex := pymsteams.NewTeamsWebhookException("public fail")
	if ex.Error() != "public fail" {
		t.Error("TeamsWebhookException error message")
	}
}

File: RepoTransBench/target_projects/Python/Java/rveachkc_pymsteams/src/test/java/com/example/pymsteams/publics/PublicCardSectionAndPotentialActionTest.java
package com.example.pymsteams.publics;

import com.example.pymsteams.CardSection;
import com.example.pymsteams.PotentialAction;
import com.example.pymsteams.TeamsWebhookException;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.*;

public class PublicCardSectionAndPotentialActionTest {

    @Test
    void testCardSectionBasicPublic() {
        CardSection section = new CardSection();
        assertSame(section, section.title("Public Section Title"));
        assertEquals("Public Section Title", section.payload.get("title"));
        assertEquals("public activity", section.activityTitle("public activity").payload.get("activityTitle"));
        assertEquals("public subtitle", section.activitySubtitle("public subtitle").payload.get("activitySubtitle"));
        assertEquals("https://example.com/img.png", section.activityImage("https://example.com/img.png").payload.get("activityImage"));
        assertEquals("public text here", section.activityText("public text here").payload.get("activityText"));
        assertEquals("greetings", section.text("greetings").payload.get("text"));
        section.linkButton("Visit", "https://visit.com");
        List<Map<String, Object>> actions = (List<Map<String, Object>>) section.payload.get("potentialAction");
        assertEquals("Visit", actions.get(0).get("name"));
        assertEquals(false, section.disableMarkdown().payload.get("markdown"));
        assertEquals(true, section.enableMarkdown().payload.get("markdown"));
        Map<String, Object> dumped = section.dumpSection();
        assertNotNull(dumped);
        assertTrue(dumped instanceof Map);
    }

    @Test
    void testCardSectionAddFactAndAddImagePublic() {
        CardSection section = new CardSection();
        section.addFact("fact_one", "value_one");
        List<Map<String, String>> facts = (List<Map<String, String>>) section.payload.get("facts");
        assertEquals("fact_one", facts.get(0).get("name"));
        assertEquals("value_one", facts.get(0).get("value"));
        section.addFact("fact_two", "value_two");
        facts = (List<Map<String, String>>) section.payload.get("facts");
        assertEquals(2, facts.size());
        section.addImage("https://img-server.com/photo1.jpg", "photo title");
        List<Map<String, Object>> images = (List<Map<String, Object>>) section.payload.get("images");
        assertEquals("photo title", images.get(0).get("title"));
        section.addImage("https://img-server.com/photo2.jpg");
        images = (List<Map<String, Object>>) section.payload.get("images");
        assertFalse(images.get(1).containsKey("title"));
    }

    @Test
    void testCardSectionFactAndImageKeysPublic() {
        CardSection section = new CardSection();
        List<Map<String, String>> facts = new ArrayList<>();
        Map<String, String> fact = new HashMap<>();
        fact.put("name", "init_name");
        fact.put("value", "init_val");
        facts.add(fact);
        section.payload.put("facts", facts);
        section.addFact("another_name", "another_val");
        List<Map<String, String>> factsResult = (List<Map<String, String>>) section.payload.get("facts");
        assertEquals(2, factsResult.size());

        List<Map<String, Object>> images = new ArrayList<>();
        Map<String, Object> image = new HashMap<>();
        image.put("image", "img_obj");
        images.add(image);
        section.payload.put("images", images);
        section.addImage("more-img-url");
        List<Map<String, Object>> imagesResult = (List<Map<String, Object>>) section.payload.get("images");
        assertEquals(2, imagesResult.size());
    }

    @Test
    void testPotentialActionInputsAndActionsPublic() {
        PotentialAction pa = new PotentialAction("PublicAction");
        pa.addInput("TextInput", "pub_input", "Public Input Title", false);
        List<Map<String, Object>> inputs = (List<Map<String, Object>>) pa.payload.get("inputs");
        assertEquals(false, inputs.get(0).get("isMultiline"));

        pa.addInput("ChoiceInput", "pub_input2", "Public Choice Input", true);
        pa.addChoice("pub_display", "pub_value");
        inputs = (List<Map<String, Object>>) pa.payload.get("inputs");
        assertTrue(inputs.get(inputs.size() - 1).containsKey("choices"));

        pa.addAction("CustomActionType", "ActionPublic", Arrays.asList("https://example.org"));
        List<Map<String, Object>> actions = (List<Map<String, Object>>) pa.payload.get("actions");
        assertEquals("CustomActionType", actions.get(0).get("@type"));
        pa.addAction("secondtype", "publicaction", Arrays.asList("weburl"), "some body here");
        actions = (List<Map<String, Object>>) pa.payload.get("actions");
        assertEquals("some body here", actions.get(1).get("body"));
    }

    @Test
    void testPotentialActionAddOpenURIAndExceptionsPublic() {
        PotentialAction pa = new PotentialAction("testopen");
        List<Map<String, Object>> targets = new ArrayList<>();
        Map<String, Object> t = new HashMap<>();
        t.put("os", "mobile");
        t.put("uri", "https://other-url.com/");
        targets.add(t);
        PotentialAction res = pa.addOpenURI("OpenOther", targets);
        assertEquals(targets, res.payload.get("targets"));

        assertThrows(IllegalArgumentException.class, () -> {
            pa.addOpenURI("Failer", 12345);
        });
    }

    @Test
    void testPotentialActionDumpPublic() {
        PotentialAction pa = new PotentialAction("dumpPA");
        Map<String, Object> dumped = pa.dumpPotentialAction();
        assertNotNull(dumped);
        assertTrue(dumped instanceof Map);
    }

    @Test
    void testTeamsWebhookExceptionReprPublic() {
        TeamsWebhookException ex = new TeamsWebhookException("public fail");
        assertTrue(ex instanceof Exception);
        assertTrue(ex.getMessage().contains("public fail"));
    }
}